File: backend/src/main/java/com/marmouset/workout/app/usecase/exercise/trained/DeleteTrainedExerciseUseCase.java
package com.marmouset.workout.app.usecase.exercise.trained;

import com.marmouset.workout.app.domain.workout.WorkoutLogNotFoundException;
import com.marmouset.workout.app.port.in.exercise.DeleteTrainedExercise;
import com.marmouset.workout.app.port.out.exercise.trained.DeleteTrainedExerciseRepoRequest;
import com.marmouset.workout.app.port.out.exercise.trained.TrainedExerciseRepository;
import com.marmouset.workout.app.port.out.workout.WorkoutLogRepository;
import java.util.UUID;
import org.springframework.stereotype.Component;

@Component
class DeleteTrainedExerciseUseCase implements DeleteTrainedExercise {
  private final TrainedExerciseRepository trainedExerciseRepository;
  private final WorkoutLogRepository workoutLogRepository;


  public DeleteTrainedExerciseUseCase(
      TrainedExerciseRepository trainedExerciseRepository,
      WorkoutLogRepository workoutLogRepository) {
    this.trainedExerciseRepository = trainedExerciseRepository;
    this.workoutLogRepository = workoutLogRepository;
  }

  @Override
  public void delete(UUID logId, Long trainedId)
      throws WorkoutLogNotFoundException {
    if (!workoutLogRepository.exists(logId)) {
      throw new WorkoutLogNotFoundException(logId);
    }
    trainedExerciseRepository.delete(
        new DeleteTrainedExerciseRepoRequest(trainedId));
  }
}

File: frontend/src/app/usecase/exercise/DeleteTrainedExerciseUseCase.ts
import DeleteTrainedExercise 
  from '../../port/in/exercise/DeleteTrainedExercise';
import TrainedExerciseClient from '../../port/out/TrainedExerciseClient';

export default class DeleteTrainedExerciseUseCase 
implements DeleteTrainedExercise {
  
  constructor(
    private readonly client: TrainedExerciseClient
  ) {
    
  }
  
  delete(workoutId: string, trainedId: number): Promise<void> {
    return this.client.delete(workoutId, trainedId);
  }
  
}